doc/docs/Python_Tutorials/Adjoint_Solver.md

```diff
@@ -2,13 +2,704 @@
 # Adjoint Solver
 ---
 
-Meep contains an adjoint-solver module for efficiently computing the gradient of an arbitrary function of the mode coefficients (S-parameters), DFT fields, and far fields (using the analytic near-to-far field transformation) with respect to $\varepsilon$ on a discrete spatial grid (a [`MaterialGrid`](../Python_User_Interface.md#materialgrid) class object) at multiple frequencies over a broad bandwidth. Regardless of the number of degrees of freedom for the grid points, just **two** separate timestepping runs are required. The first run is the "forward" calculation to compute the objective function and the DFT fields of the design region. The second run is the "adjoint" calculation to compute the gradient of the objective function with respect to the design variables. The adjoint run involves a special type of source distribution used to compute the DFT fields of the design region. The gradient is computed in post processing using the DFT fields from the forward and adjoint runs. This is all done automatically, and is described in:
+Meep contains an adjoint-solver module for efficiently computing the
+gradient of an arbitrary function of the mode coefficients
+($\mathcal{S}$-parameters), DFT fields, local density of states
+(LDOS), and far fields (Fraunhofer condition) with respect to
+$\varepsilon$ on a discrete spatial grid (a
+[`MaterialGrid`](../Python_User_Interface.md#materialgrid) class
+object) at multiple frequencies over a broad bandwidth. Regardless of
+the number of degrees of freedom for the grid points, just **two**
+distinct timestepping runs are required. The first run is the
+"forward" calculation to compute the objective function and the DFT
+fields of the design region. The second run is the "adjoint"
+calculation to compute the gradient of the objective function with
+respect to the design variables. The adjoint run involves a special
+type of current source distribution used to compute the DFT fields of
+the design region. The gradient is computed in post processing using
+the DFT fields from the forward and adjoint runs. The gradient
+calculation is fully automated. The theoretical and computational
+details of the adjoint-solver module are described in this manuscript:
```

Comment:
nit: paper or publication, not manuscript (it's already been peer reviewed and accepted).

Reply:
Fixed.